paimon-core/src/main/java/org/apache/paimon/metrics/Metrics.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.metrics;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Core of Paimon metrics system. */
+public class Metrics {
+    private static volatile Metrics instance;
+
+    private Metrics() {}
+
+    public static Metrics getInstance() {
+        if (instance == null) {
+            synchronized (Metrics.class) {
+                if (instance == null) {
+                    instance = new Metrics();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /** The metrics groups. */
+    private final List<MetricGroup> metricGroups = new ArrayList<>();
+
+    /** Add a metric group. */
+    public void addGroup(AbstractMetricGroup group) {
+        metricGroups.add(group);
+    }
+
+    /** Get metric groups. */
+    public List<MetricGroup> getMetricGroups() {
+        return metricGroups;
+    }
```

Comment:
You use `synchronized` to protect `getInstance` from multi-threading. What about `metricGroups`? Will it be visited by multiple threads?

Reply:
Thanks a lot for reminding, it's mistake here, I've fixed it.